cmd/fyne/internal/commands/package.go

```diff
@@ -338,8 +338,16 @@ func (p *Packager) validate() error {
 		_, _ = fmt.Fprint(os.Stderr, "Parameter -executable is ignored for mobile builds.\n")
 	}
 
-	if p.name == "" || p.os == "wasm" || p.os == "gopherjs" || p.os == "web" {
-		p.name = exeName
+	if p.os == "wasm" || p.os == "gopherjs" || p.os == "web" {
+		if p.name == "" {
```

Comment:
I think this logic has broken the case where there is no name set, but the OS is not one of the web targets...
It should default to the exeName as it used to.